Added note about ASPNETCORE projects (#13490)
dotnet/docs: includes/core-testing-note-aspnet.md

> This article is about testing a .NET Core project. If you're testing an **ASP.NET Core** project, see [Integration tests in ASP.NET Core](/aspnet/core/test/integration-tests?view=aspnetcore-2.2#test-app-prerequisites).

Reviewer: ```suggestion
This article is about testing a .NET Core project. If you're testing an **ASP.NET Core** project, see [Integration tests in ASP.NET Core](/aspnet/core/test/integration-tests#test-app-prerequisites).
```

>[!IMPORTANT]

> >If you're _not_ using the **.NET Core 3.0 SDK**, you must add two additional package references, `Microsoft.AspNetCore.App` and `Microsoft.AspNetCore.Mvc.Testing`.

Reviewer 2: You will need to add the Microsoft.AspNetCore.App if TargetFramework is less than netcoreapp3.0, even if you are building with the 3.0 SDK.

>
>```

> >dotnet add package Microsoft.AspNetCore.App

Reviewer 2: This should be added without a package version. Does `dotnet add package` respect that? I imagine you'd get a version and then get a warning in the build.

Author: It adds whatever version is the latest that it uses. I can't see an easy way to force it to versionless apart from editing the csproj file directly.

Reviewer 2: cc @dsplaisted. We should be careful recommending this, then. 

I forget what happens when you have a version, does it win or just warn?

Reviewer 3: It should warn

Reviewer 2: warn and then ignore or warn and then use what you say?

> >dotnet add package Microsoft.AspNetCore.Mvc.Testing

Reviewer 2: This is the first time I've seen this package as a requirement here. Is it always required?

Author: According to https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.2#test-app-prerequisites it's required.

Author: After reading a bit more about this, I believe you're correct. This package is only needed to supplement MVC testing which isn't pertinent to this note.

>[!IMPORTANT]

> >If you're testing an ASP.NET application or library, and you are targeting a framework _other_ than `netcoreapp3.0`, you must add an additional package reference: `Microsoft.AspNetCore.App`.

Reviewer: I think this sentence needs some clarification. `netcoreapp3.0` is the ONLY valid TFM as of ASP.NET Core 3.0. ASP.NET Core 2.2 and earlier versions allow you to target either .NET Framework or .NET Core.

```suggestion
>If you're testing an ASP.NET Core application, and you're targeting a framework _other_ than `netcoreapp3.0`, you must add an additional package reference: `Microsoft.AspNetCore.App`.
```
